.github/actions/prepare_poetry_env/action.yml

```diff
@@ -0,0 +1,19 @@
+name: 'Prepare Poetry environment'
+description: 'This composite actions checks out out the project, installs Poetry, and install the project in the Poetry environment'
+inputs:
+  python-version:
+    description: 'The Python version to use'
+    required: true
+    default: '3.10'
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/setup-python@v5
+      with:
+        python-version: ${{ inputs.python-version }}
```

Comment:
I removed the custom action and started to use `exasol/python-toolbox/.github/actions/python-environment@0.12.0`
As soon as our python-toolbox GH actions supports caching, it will work out of the box here.